tests/unit/lms/services/oauth_http_test.py

```diff
@@ -36,24 +37,104 @@ def test_convenience_methods(self, method, svc, http_service, oauth2_token_servi
         )
         assert response == http_service.request.return_value
 
-    def test_it_crashes_if_theres_already_an_Authorization_header(self, svc):
+    def test_request_crashes_if_theres_already_an_Authorization_header(self, svc):
         with pytest.raises(AssertionError):
             svc.request(sentinel.method, sentinel.url, headers={"Authorization": "foo"})
 
-    def test_it_raises_if_theres_no_access_token_for_the_user(
+    def test_request_raises_if_theres_no_access_token_for_the_user(
         self, svc, oauth2_token_service
     ):
         oauth2_token_service.get.side_effect = OAuth2TokenError
 
         with pytest.raises(OAuth2TokenError):
             svc.request(sentinel.method, sentinel.url)
 
-    def test_it_raises_if_the_HTTP_request_fails(self, svc, http_service):
+    def test_request_raises_if_the_HTTP_request_fails(self, svc, http_service):
         http_service.request.side_effect = HTTPError
 
         with pytest.raises(HTTPError):
             svc.request(sentinel.method, sentinel.url)
 
+    def test_access_token_request(
+        self,
+        svc,
+        http_service,
+        OAuthTokenResponseSchema,
+        oauth_token_response_schema,
+        oauth2_token_service,
+    ):
+        oauth_token_response_schema.parse.return_value = {
+            "access_token": "access_token",
+            "refresh_token": "refresh_token",
+            "expires_in": 1234,
+        }
+
+        svc.access_token_request(
+            sentinel.token_url,
+            sentinel.redirect_uri,
+            sentinel.auth,
+            sentinel.authorization_code,
+        )
+
+        http_service.post.assert_called_once_with(
+            sentinel.token_url,
+            data={
+                "grant_type": "authorization_code",
+                "redirect_uri": sentinel.redirect_uri,
+                "code": sentinel.authorization_code,
+            },
+            auth=sentinel.auth,
+        )
+        OAuthTokenResponseSchema.assert_called_once_with(http_service.post.return_value)
+        oauth2_token_service.save.assert_called_once_with(
+            "access_token", "refresh_token", 1234
+        )
+
+    def test_access_token_request_raises_HTTPError_if_the_HTTP_request_fails(
+        self, svc, http_service
+    ):
+        http_service.post.side_effect = HTTPError
+
+        with pytest.raises(HTTPError):
+            svc.access_token_request(
+                sentinel.token_url,
+                sentinel.redirect_uri,
+                sentinel.auth,
+                sentinel.authorization_code,
+            )
```

Comment:
A caller pattern might be appropriate here to cut down on the waffle? We explicitly don't care what the method was called with in these failure cases.

Reply:
Done: https://github.com/hypothesis/lms/pull/2987/commits/1a1064e34fe3d19d642560117aae89ccb020b1f9